src/adagrams.js

```diff
@@ -1,15 +1,136 @@
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  let weightedList = []
+  for (let key of Object.keys(LETTER_POOL)) {
+      const letterList = new Array(LETTER_POOL[key])
+      letterList.fill(key,0)
+      weightedList.push(...letterList)
+  }
+  const lettersList = [];
+  const letterDrawn = {};
+
+  while (lettersList.length <10) {
+    let randomNum = weightedList[Math.floor(Math.random()*weightedList.length)];
+    if (randomNum in letterDrawn){
+      if (letterDrawn[randomNum] < LETTER_POOL[randomNum]) {
+        letterDrawn[randomNum] +=1
+      } else {
+        continue
+      }
+    } else {
+      letterDrawn[randomNum] = 1
+    }
+    lettersList.push(randomNum);
+  }
+  return lettersList;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const lettersBank = {};
+  for (let letter of lettersInHand) {
+    if (letter in lettersBank) {
+      lettersBank[letter] +=1
+    } else {
+      lettersBank[letter] =1
+    }
+  }
+  const capitialInput = input.toUpperCase();
+  for (let letter of capitialInput) {
+      if (lettersBank[letter] >=1) {
+          lettersBank[letter] -=1
+      } else {
+          return false;
+      }
```

Comment:
Great work getting comfortable with JavaScript's syntax. It looks like you're getting use to looping over data and adding conditional logic ✅